### PR TITLE
added automatic client page reload

### PR DESCRIPTION
**Pull request by khassel**
solution for #3105 

~~not sure if updatenotification is the right place, so opinions?~~

now impleneted in `main.js`

### CODE REVIEW

**Comment by bugsounet** [comment]
Hi, not sure it's better place
because some user disable updatenotification and it's not the fonction of it :/

For me, maybe best place is [main.js](https://github.com/MichMich/MagicMirror/blob/master/js/main.js#L64-L66)
when all promise of all DOMs are resolved (All DOM_OBJECTS_CREATED of all modules)
I consider that all modules are ready and you code can start from line 67 of main.js

In this case `checkServerInterval` will be a global feature and not a module feature

File: js/main.js [hunk 1]
```diff
@@ -568,12 +568,33 @@ const MM = (function () {
 		 */
 		modulesStarted: function (moduleObjects) {
 			modules = [];
+			let startUp = "";
+
 			moduleObjects.forEach((module) => modules.push(module));
 
 			Log.info("All modules started!");
 			sendNotification("ALL_MODULES_STARTED");
```
Comment (bugsounet) [inline js/main.js:576]: I no longer thought about “ALL_MODULES_STARTED”

it makes more sense ;)

File: tests/e2e/helpers/mock-console.js
Comment (rejas) [inline tests/e2e/helpers/mock-console.js:1]: Are those changes here related to the PR or just general cleanups?

File: js/main.js (hunk 1 above)
Comment (khassel) [reply js/main.js:576]: @bugsounet is this an "o.k." or a change request?

File: tests/e2e/helpers/mock-console.js
Comment (khassel) [reply tests/e2e/helpers/mock-console.js:1]: removed test changes with latest commit

File: js/defaults.js [hunk 2]
```diff
@@ -29,6 +29,11 @@ const defaults = {
 	// e.g. you need to add `frameguard: false` for embedding MagicMirror in another website, see https://github.com/MichMich/MagicMirror/issues/2847
 	httpHeaders: { contentSecurityPolicy: false, crossOriginOpenerPolicy: false, crossOriginEmbedderPolicy: false, crossOriginResourcePolicy: false, originAgentCluster: false },
 
+	// properties for checking if server is alive and has same startup-timestamp, the check is per default enabled
+	// (interval 30 seconds). If startup-timestamp has changed the client reloads the magicmirror webpage.
+	checkServerInterval: 30 * 1000,
+	reloadAfterServerRestart: true,
```
[the remark this answers is not in the document]
Comment (khassel) [reply js/defaults.js]: changed it to false

File: js/main.js (hunk 1 above)
Comment (bugsounet) [reply js/main.js:576]: It's ok for me now.
 I have says that before your change to main.js

**Review by rejas — APPROVED** [review]

**Pull request merged**